W-14659026-SFDC-Bridge-se (#359)
mulesoft/docs-rpa: rpa-home/modules/ROOT/pages/invoke-rpa-from-flow-build.adoc

# Build an Automation

Implement your automation in RPA Builder.

> ## Before You Begin

Reviewer: This heading does not go with the wording in the first bullet. I think that this topic should be written differently.

Author: Replaced wording:

Author: * For input and output parameters of a process, only use variable types that are supported in Salesforce Invocable Actions as RPA Builder Activity Parameters:
+
The following table shows possible mappings:

> * The following variable types of RPA Builder Activity Parameters are supported as input or output variables of an Invocable Action by Salesforce:

Reviewer: ```suggestion
* Salesforce supports the following types of RPA Builder activity parameters as invocable action input or output variables:
```
Global: Do you have to initial-cap Activity Parameters and Invocable Actions? Does Salesforce support activity parameter types as invocable action variables or activity parameters as invocable action variables?

Author: Changed the wording to fit the heading.

Author: * For input and output parameters of a process, only use variable types that are supported in Salesforce Invocable Actions as RPA Builder Activity Parameters:
+
The following table shows possible mappings:

+
[cols="1,1,2"]
|===
|*Activity Parameter Type*
|*Input/Output Variable Type*
|*Additional Information*

|Alphanumeric
|ID, Long, String

> |You can use comma-seperated strings within strings to replace an array. In this case, name the variable in plural and add the suffix CommaSeperatedList to the variable names; for examplespecialAccountsCommaSeperatedList.

Reviewer: Required now

```suggestion
|You can use comma-seperated strings within strings to replace an array. In this case, pluralize the variable names and add the suffix `CommaSeparatedList` to the names; for example, `specialAccountsCommaSeparatedList`.
```

Reviewer: In the above text, "Seperated" is spelled wrong. Hopefully, this isn't in the product. If so, it should be corrected.

|Integer
|Integer
| 

|Floating-point
|Decimal, Double
| 

|Boolean
|Boolean
| 
|===

> * Name your Activity Parameters wisely:

Reviewer: ```suggestion

```

Reviewer: The first sentence is not necessary. It duplicates the first bullet in meaning. If you remove this, everything else can move up a level. (I marked this.)

> ** Use meaningful names to quickly find your input and output resources in Salesforce Flow.

Reviewer: ```suggestion
* Use meaningful names to quickly find your input and output resources in Salesforce Flow.
```

> ** To implement Salesforce naming standards, adhere to the following rules:

Reviewer: ```suggestion
+
To implement Salesforce naming standards, adhere to the following rules:
```

> *** Use camelCase.

Reviewer: Required now

```suggestion
** Use camel case.
```

> *** Use only the characters `a-z`, `A-Z` and `0-9`.

Reviewer: ```suggestion
** Use only the characters `a-z`, `A-Z` and `0-9`.
```

> *** Restrict the length of the names to 255 characters.

Reviewer: ```suggestion
** Restrict the length of the names to 255 characters.
```

## Build an Automation

> Refer to https://docs.mulesoft.com/rpa-manager/processautomation-develop#build-a-process to learn how to build processes.

Reviewer: Required now

Why bold?

Author: That is strange. It's not bolded and does not appear bold in the local build...
